### PR TITLE
chore(README): add architecture diagram

### PR DESCRIPTION
**Pull request by xilosada**
Screenshot 2024-06-26 at 11 34 17

## Summary by CodeRabbit

- **Documentation**
  - Added an architecture diagram using Mermaid syntax to the README file, visualizing system components and their relationships.

### CODE REVIEW

File: README.mdx [hunk 1]
```diff
@@ -71,6 +71,71 @@ Calimero Network's story is one of relentless pursuit of a more private,
 efficient, and user-centric decentralized web, setting a new standard for what
 decentralized platforms can achieve.
 
+## Architecture
+
+```mermaid
+graph TD
+    subgraph Datastore
+        DS1[Context]
+        DS2[General]
```
Comment (miraclx) [inline README.mdx]:
```suggestion
        DS2[Generic]
```

File: README.mdx [hunk 2]
```diff
@@ -71,6 +71,71 @@ Calimero Network's story is one of relentless pursuit of a more private,
...
+    subgraph Blobstore
+        BS1[FS]
+        BS2[S3]
+        BS3[Applications]
+        BS4[Generic Data]
+        BS3 --> BS1
+        BS4 --> BS2
+    end
+
+    Network --> Node
```
Comment (miraclx) [inline README.mdx]:
```suggestion
    Node --> Network
```

File: README.mdx [hunk 3]
```diff
@@ -71,6 +71,71 @@ Calimero Network's story is one of relentless pursuit of a more private,
...
+
+    subgraph Client
+        C1[C-API]
+    end
+
+    subgraph Server
+        S1[Auth Gated Client]
+        S2[CLI]
```
Comment (miraclx) [inline README.mdx]: can we pull this out? it depends on server, not part of it

File: README.mdx [hunk 4]
```diff
@@ -71,6 +71,71 @@ Calimero Network's story is one of relentless pursuit of a more private,
...
+        end
+    end
+
+    subgraph Blobstore
+        BS1[FS]
+        BS2[S3]
+        BS3[Applications]
+        BS4[Generic Data]
+        BS3 --> BS1
+        BS4 --> BS2
```
Comment (miraclx) [inline README.mdx]: the File system & S3 are examples of blobstore implementations

while Applications & Generic Data are examples of what's stored inside the blobstore

File: README.mdx [hunk 5]
```diff
@@ -71,6 +71,71 @@ Calimero Network's story is one of relentless pursuit of a more private,
...
+        DS2 --> DS7[Scope]
+        DS2 --> DS8[Fragment]
+        DS3 --> DS9[KAD]
+
+        subgraph impl
+          DI1[Memory]
+          DI2[S3]
+          DI3[Level]
+          DI4[RocksDB]
+        end
```
Comment (miraclx) [inline README.mdx]: like this, for example

**Review by miraclx — COMMENTED** [review]
love this, let's just fix the arrow directions and some other small things

thank you!!!

**Comment by xilosada** [comment]
@miraclx 
```mermaid
graph TD

    S2[CLI]

    subgraph Datastore
        DS1[Context]
        DS2[Generic]
        DS3[Network]
        DS1 --> DS4[Identity]
        DS1 --> DS5[State]
        DS1 --> DS6[Transactions]
        DS2 --> DS7[Scope]
        DS2 --> DS8[Fragment]
        DS3 --> DS9[KAD]

        subgraph \n\n
          DI1[Memory]
          DI2[S3]
          DI3[Level]
          DI4[RocksDB]
        end
    end

    subgraph Blobstore
        BS3[Applications]
        BS4[Generic Data]
        subgraph \n
          BS1[FS]
          BS2[S3]
       end
    end
    Blobstore --> BS3

    Node --> Network

    subgraph Client
        C1[C-API]
    end

    subgraph Server
        S1[Auth Gated Client]
        S3[JSONRPC]
        S4[WebSocket]
        S5[Raw Sockets]

        Apps --> S3
        Apps --> S4
        Apps --> S5
    end
    S2 --> Server
    S1 --> Client
    Node --> Runtime[Runtime]
    Node --> Datastore
    Node --> Blobstore
    C1 --> Node
    Node --> S1

    subgraph Apps
    end

```

File: README.mdx [hunk 6]
```diff
@@ -71,6 +71,71 @@ Calimero Network's story is one of relentless pursuit of a more private,
...
+    Blobstore --> BS3
+
+    Node --> Network
+
+    subgraph Client
+        C1[C-API]
+    end
+
+    subgraph Server
+        S1[Auth Gated Client]
```
Comment (miraclx) [inline README.mdx]: this isn't inside server, it's own thing

- `server` depends on `auth gated client`
- `CLI` depends on `server` and `auth gated client`
- `auth gated client` depends on `client`
- `client` depends on `node`
- `scope` depends on `fragment`

Comment (miraclx) [reply README.mdx]: `C-API` depends on `client`

Comment (miraclx) [reply README.mdx]: If you can represent this in mermaid:

- `node` depends on `context` and `network` inside `datastore`
- `auth gated client` depends on `generic` inside `datastore`

**Comment by miraclx** [comment]
took a stab at this:

```mermaid
graph TD

    subgraph Datastore
        ctx[Context]
        gen[Generic]
        net[Network]
        ctx --> id[Identity]
        ctx --> st[State]
        ctx --> tx[Transactions]
        gen --> sc[Scope]
        gen --> fr[Fragment]
        net --> kad[KAD]

        subgraph " "
            direction LR
            mem[Memory]
            s3[S3]
            lvl[Level]
            rdb[RocksDB]
        end
    end

    subgraph Blobstore
        apps[Applications]
        genData[Generic Data]
        
        subgraph " "
            direction LR
            fs[FS]
            blobS3[S3]
        end
    end
    
    subgraph Server
        direction LR
        jrpc[JSONRPC]
        ws[WebSocket]
        raw[Raw Sockets]
    end

    applications[Apps]
    client[Client]
    capi[C-API]
    auth[Auth Gated Client]
    did[DID]
    cli[CLI]
    node[Node]
    network[Network]
    runtime[Runtime]

    client --> node

    applications --> capi
    applications --> client
    applications --> Server

    cli --> client
    cli --> auth
    cli --> Server

    Server --> auth

    capi --> client
    
    auth --> client
    auth --> did
    did --> gen

    node --> network
    node --> runtime
    node --> ctx
    node --> net
    node --> Blobstore
```

**Review by miraclx — APPROVED** [review]
lgtm..

**Pull request merged**